agent/agent.go

```diff
@@ -328,6 +330,9 @@ func (a *Agent) Run(shutdown chan struct{}) error {
 	}
 	ticker := time.NewTicker(a.Config.Agent.Interval.Duration)
 
+	ent := enterprise.NewEnterprise(a.Config.Agent.Enterprise, a.Config.Agent.Hostname, a.Version, shutdown)
+	go ent.Open()
```

Comment:
It checks the config to see if we should perform registration with enterprise and startup the admin interface. If there are no configured enterprise hosts, it just returns.